docs/admin/troubleshooting.md

```diff
@@ -522,14 +522,50 @@ To resolve the issue:
 - Start all `daos_server` processes.
 - Verify that all ranks were able to re-join via `dmg system query`.
 
+### Running out of Memory during Rebuild
+
+Several environment variables can be set in the env_vars section of the yaml file of the DAOS server
+to control how much memory will be used by the migration service to transfer data while handling
+a rebuild, reintegrate, drain or rebalancing operation.
+
+The D\_MIGRATE\_TGT\_ULT\_CNT and D\_MIGRATE\_SYS\_ULT\_CNT environment variables control how many
+user level threads (i.e. ULT) will be created at most in the engine while processing one of the
+operation listed above. D\_MIGRATE\_TGT\_ULT\_CNT is for the total number of ULTs across the target
+xstreams while D\_MIGRATE\_SYS\_ULT\_CNT is for the system xstream in charge of object scanning.
+Each ULT will consume a stack size of 128KiB. To sum up, the following formula defines how much
+memory will be allocaged at most for all the ULTs in an engine to manage data migration:
+
+```
+(D_MIGRATE_TGT_ULT_CNT + D_MIGRATE_SYS_ULT_CNT) x 128KiB
+```
+
+The default value for both environment variables is 4096. This gives us a 1GB of memory.
+
+Moreover, the D\_MIGRATE\_TGT\_MEM\_SIZE and D\_MIGRATE\_SYS\_MEM\_SIZE environment variables
```

Comment:
Now the static macro "MIGRATE_MAX_SIZE" is to control the total inflight rebuild data size across all VOS tgt XS, currently the default value set as 25% of system DMA buffer size.
One point is now rebuild possibly generate a few large dkey migrate ULT (migrate_one_ult()) with data size exceed the limit in that case we have to make it pass to avoid the rebuild task cannot complete (https://github.com/daos-stack/daos/pull/14052 fixed a related bug).

This one looks need not to separately define two ENVs for  SYS and TGT either, the number of tls->mpt_inflight_size/tls->mpt_inflight_max_size is not really useful on SYS XS.